### ISSUE TITLE
File transfer speed is very slow

### ISSUE THREAD

**Issue by VxMxPx** [andyholmes/gnome-shell-extension-gsconnect#214]
Copying files from computer to phone, goes very slow, we're talking dial-up modem type of slow. A normal size MP3 file or an image, would usually just hang / froze Nautilus. 

Copying files from phone to computer, is completely normal though.

**Comment by AdamKane41** [comment]
also fast on my end using share button via sys tray

**Comment by andyholmes** [comment]
I guess it's worth pointing out that in v13 (`master`), the "direct share" used in the device menu and Nautilus has been improved a lot. Basically it just "attaches" the file to a connection and let's the remote device transfer it as fast as it can (in a dedicate thread). Previously it used to send a ~4kb chunk in a thread and return, then repeat until finished.

`sshfs` is a little different, since it relies on local FUSE performance and the remote device's response time. Nothing much has changed with this in v13, since it's mostly sshfs to do the transferring.

**Comment by VxMxPx** [comment]
This is a problem when using `sshfs` (Browse), I've tried share now and that's more interesting... It goes a little bit faster, but then just randomly fails (on the phone message is shown: Failed to receiving file).

If this is of any interest:

```
› sshfs --version
SSHFS version 3.5.0
FUSE library version 3.2.5
using FUSE kernel interface version 7.26
fusermount3 version: 3.2.5
```

```
› uname -a
Linux pc-mistik 4.18.10-arch1-1-ARCH #1 SMP PREEMPT Wed Sep 26 09:48:22 UTC 2018 x86_64 GNU/Linux
```

```
› grep version .local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/metadata.json 
"version": "12",
```

I'll try to upgrade extension to v13 from master.

**Comment by andyholmes** [comment]
Are you by chance using Android Oreo?

**Comment by VxMxPx** [comment]
Yes, It's Android version 8.0.0.

**Comment by andyholmes** [comment]
Ah, I think you need a special build (right now) for Oreo because of the "peristent notification" & background service changes. One sec and I'll see if I can build you one.

**Comment by VxMxPx** [comment]
Thanks for this. However when I install it on my phone, the WIFI stops working (I've tried twice, and it happened both times). 

I've upgraded extension to the version 13 and manage to connect with original version of KDE Connect, but transfer speeds are still the same. 

It was working really fine a while back, then some upgrades happened (wish I'd knew what and when) and since then it's really slow.

I've reinstalled Arch yesterday (unrelated problems), and it's still the same. Really interesting that no one else is experiencing this.

Anyhow I'll look around a little bit namely sshfs, and KDE Connect and if I find any solution I'll report back.
